RHDHBUGS-3014: Create Guidance for Enforcing Unique Workflow IDs (#2123)
redhat-developer/red-hat-developers-documentation-rhdh: modules/extend_orchestrator-in-rhdh/proc-restore-workflow-visibility-by-removing-duplicate-entries.adoc

.. For the new workflow version, modify the `id` field to include a version identifier:
+
[source,yaml]
----
id: customer-onboarding-v2
version: "2.0"
name: Customer Onboarding
----
.. Maintain the original workflow ID for the current deployment.
.. Build and deploy the updated workflow definition.

. Remove outdated workflow deployments:
.. After confirming the new workflow operates correctly, remove the old workflow deployment.
.. Verify that all instances of the old workflow have completed.
.. Delete the workflow resources from your cluster:
+
[source,bash]
----
oc delete sonataflow <old-workflow-name> -n <workflow-namespace>
----
+
[NOTE]
====
Deleting the workflow deployment removes it from the cluster but preserves historical execution records in the Data Index. Users can still view past workflow runs in {product-very-short}.
====

. Clean historical data if necessary:
+
If duplicate entries persist in the UI after you remove the workflow deployments, the Data Index database has historical records from earlier workflow executions. These records preserve the execution history of workflows that ran on different runtime servers over time.
+
[IMPORTANT]
====

> Removing historical data from the Data Index database can affect your ability to view past workflow execution records. Consider backing up the data before removal.

Reviewer: ```suggestion
Back up your workflow execution records before you remove historical data from the Data Index database. Removing this data permanently prevents access to past execution records.
```

====
+
.. Connect to the Data Index database to verify the duplicate entries.
.. Query the workflow definitions to identify duplicate entries:
+
[source,sql]
----
SET search_path TO "sonataflow-platform-data-index-service";
SELECT id, version, name FROM definitions;
----
.. Evaluate whether to remove the historical data.
You can keep the historical data to retain past workflow execution records, which allows you to view the execution history and results of completed workflow instances.

> Alternatively, contact your system administrator or Red Hat support for guidance on safely removing historical duplicate entries from the Data Index without affecting active workflow operations.

Reviewer: I believe this should have capital S, but oddly I can't find it in the style guide.  I found this [instance](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/configuring_basic_system_settings/assembly_accessing-the-red-hat-support_configuring-basic-system-settings) in Linux doc, bit nothing in RHDH doc.

```suggestion
Alternatively, contact your system administrator or Red Hat Support for guidance on safely removing historical duplicate entries from the Data Index without affecting active workflow operations.
```

> Alternatively, contact your system administrator or Red Hat Support for guidance on safely removing historical duplicate entries from the Data Index without affecting active workflow operations.

Reviewer 2: ```suggestion
Alternatively, contact your system administrator or {company-name} Support for guidance on safely removing historical duplicate entries from the Data Index without affecting active workflow operations.
```
